File: 2470-number-of-subarrays-with-lcm-equal-to-k/2470-number-of-subarrays-with-lcm-equal-to-k.py
class Solution:
    def subarrayLCM(self, nums: List[int], k: int) -> int:
        ans = 0
        for i in range ( len(nums)):
            x = nums[i]
            if(x==k):
                ans+=1
            for j in range(i+1,len(nums)):
                x = math.lcm(x,nums[j])
                if(x==k):
                    ans+=1
                if(x>k):
                    break
        return ans

File: 2470-number-of-subarrays-with-lcm-equal-to-k/2470-number-of-subarrays-with-lcm-equal-to-k.cpp
class Solution {
public:
    int subarrayLCM(vector<int>& nums, int k) {
        int ans =0;
        for(int i=0;i<nums.size();++i){
            long long x = nums[i];
            if(x==k)++ans;
            for(int j=i+1;j<nums.size();++j){
                x = lcm(nums[j],x);
                 // cout<<x<<" "<<i<<" "<<j<<endl;
                if(x==k){
                    ++ans;
                }
                if(x>k)break;
            }
        }
        return ans;
    }
};